docs/learn/prompts.md

```diff
@@ -31,9 +53,13 @@ print(prompt)
 # > Recommend a fantasy book
 ```
 
-For your convenience, we will also automatically dedent and strip the prompt template. This makes writing multi-line prompts a breeze:
+In this example, the `@prompt_template` decorator defines the template string `"Recommend a {genre} book"`. The genre attribute of the `BookRecommendationPrompt` class corresponds to the `{genre}` placeholder in the template. When you create an instance of the class with `genre="fantasy"`, this value is automatically injected into the template.
 
-```python
+We've implemented the `__str__` method in `BasePrompt` for easy verification of how variables are templated.
+
+For your convenience, we automatically dedent and strip the prompt template. This makes writing multi-line prompts a breeze:
+
+```python hl_lines="5-8 11"
```

Comment:
so in line with my other comment, i think after the very first example showing `BasePrompt` we should use the `prompt_template` functional approach going forward until the end where we'll cover `BasePrompt` in more detail. Originally the reason I used `BasePrompt` for all of these examples was that adding `prompt_template` didn't do anything, but since we've implemented that feature and now the decorated function is useful, we should instead just use the functional approach for these explanations?